crates/types/src/blockchain/header.rs

```diff
@@ -51,12 +45,11 @@ pub enum BlockHeader {
 }
 
 /// Helpful methods for all variants of the block header.
-#[enum_dispatch]
-pub(crate) trait GetBlockHeaderFields {
+pub trait GetBlockHeaderFields<GAF> {
```

Comment:
I don't see what is the use case of having a trait for getters(and setters in the case of the tests). 

Because it is generic, it means that it will be implemented only for `HeaderV1` and `HeaderV2` only. It can't be implemented for the `BlockHeader` enum, so it can't be used with it. Then, why do we need it? Looks like just public getters from `impl` section should be enough.

Reply:
leftover code from enum_dispatch land, will fix

Reply:
addressed in [28a816a](https://github.com/FuelLabs/fuel-core/pull/2678/commits/28a816a64b6f5e4bae82f3b1941dd5ed6dad3932)